src/arguments/CoreBoolean.ts

```diff
@@ -9,10 +9,10 @@ export class CoreArgument extends Argument<boolean> {
 
 	public run(parameter: string, context: { readonly truths?: string[]; falses?: readonly string[] } & Argument.Context): Argument.Result<boolean> {
 		const resolved = resolveBoolean(parameter, { truths: context.truths, falses: context.falses });
-		if (resolved.success) return this.ok(resolved.value);
+		if (resolved.isOk()) return this.ok(resolved.unwrap());
 		return this.error({
 			parameter,
-			identifier: resolved.error,
+			identifier: resolved.unwrapErr(),
 			message: 'The argument did not resolve to a boolean.',
 			context
 		});
```

Comment:
`isOk()` with `unwrap()` and `unwrapErr()` is an antipattern, please prefer map or match methods instead:

```typescript
return resolved.match({
	ok: (value) => this.ok(value),
	err: (identifier) => this.error({
		parameter,
		identifier,
		message: 'The argument did not resolve to a boolean.',
		context
	})
});
```

This applies to everywhere else that does the same pattern.

A better alternative, though, would be to make a new method, `errorContext`, which creates the `ArgumentError` without wrapping it inside `Err`, this way, this code would be possible:

```typescript
return resolved
	.mapErr((identifier) => this.errorContext({
		parameter,
		identifier,
		message: 'The argument did not resolve to a boolean.',
		context
	}));
```

Which avoids creating a new `Ok` object on success, as well as a function call.